skimage/measure/fit.py

```diff
@@ -837,7 +834,11 @@ def ransac(data, model_class, min_samples, residual_threshold,
     # estimate model for current random sample set
     model = model_class()
 
-    for num_trials in range(max_trials):
+    num_trials = 0
+    # max_trials can be updated inside the loop, so this cannot be a for-loop
```

Comment:
Does num_trials get updated inside the loop? Doesn't look like it, so this was probably fine?

Reply:
@stefanv yes, num_trials is updated at line 840.

```
839 while num_trials < max_trials:
840    num_trials += 1
```

Reply:
Yes, but if we use the old code that line is not there or necessary.

Reply:
In my opinion, since max_trials can be changed in the new code(because of the result of _dynamic_max_trials can update max_trials), we need to record and increase num_trials separately, so we need that row.
Am I misunderstanding your intentions?😢

Reply:
Ah, no, that makes sense! I was looking at num_trials